services-camel/src/test/java/edu/unc/lib/dl/services/camel/destroyDerivatives/DestroyDerivativesRouterIT.java

```diff
@@ -0,0 +1,345 @@
+    }
+
+    @Test
+    public void destroyTextTest() throws Exception {
+        WorkObject work = repoObjectFactory.createWorkObject(null);
+        String mimetype = "text/plain";
+        FileObject fileObj = addFileToWork(work, mimetype);
+        work.addMember(fileObj);
+
+        treeIndexer.indexAll(baseAddress);
+
+        initMarkForDeletionJob(work.getPid());
+        deletionJob.run();
+        initializeDestroyJob(Collections.singletonList(fileObj.getPid()));
+        destroyJob.run();
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor).process(any(Exchange.class));
+    }
+
+    @Test
+    public void invalidTypeTest() throws Exception {
+        WorkObject work = repoObjectFactory.createWorkObject(null);
+        FileObject fileObj = addFileToWork(work, "application/octet-stream");
+        work.addMember(fileObj);
+
+        treeIndexer.indexAll(baseAddress);
+
+        initMarkForDeletionJob(fileObj.getPid());
+        deletionJob.run();
+        initializeDestroyJob(Collections.singletonList(fileObj.getPid()));
+        destroyJob.run();
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    private FileObject addFileToWork(WorkObject work, String mimetype) throws Exception {
+        FolderObject folder = repoObjectFactory.createFolderObject(null);
+        collection.addMember(folder);
+        folder.addMember(work);
+
+        String bodyString = "Content";
+        Path storagePath = Paths.get(locationManager.getStorageLocationById(LOC1_ID).getStorageUri(work.getPid()));
+        Files.createDirectories(storagePath);
+        File contentFile = Files.createTempFile(storagePath, "file", ".txt").toFile();
+        String sha1 = "4f9be057f0ea5d2ba72fd2c810e8d7b9aa98b469";
+        String filename = contentFile.getName();
+        FileUtils.writeStringToFile(contentFile, bodyString, "UTF-8");
+
+        return work.addDataFile(contentFile.toURI(), filename, mimetype, sha1, null);
+    }
+
+    private void initMarkForDeletionJob(PID pid) {
+        deletionJob =  new MarkForDeletionJob(pid, "", agent, repoObjLoader,
+                sparqlUpdateService, aclService);
+    }
+
+    private void initializeDestroyJob(List<PID> objsToDestroy) {
```

Comment:
Totally optional, but if you don't want to have to initialize lists in order to call this method, you can do `initializeDestroyJob(PID... objsToDestroy)` in which case you'll get an array of PIDs, which you can stream with `Arrays.stream(array)`. To call it you do `initializeDestroyJob(pid1, pid2, etc);`

You could probably also make the method into `destroyObject` and immediately execute the job.